pkg/resources/management.cattle.io/v3/setting/validator.go

```diff
@@ -89,48 +89,84 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
+	valueFieldPath   = field.NewPath("value")
+)
+
 func (a *admitter) validateUserRetentionSettings(s *v3.Setting) error {
 	var err error
 
 	switch s.Name {
 	case "disable-inactive-user-after":
+		if s.Default != "" {
+			if _, fieldErr := validateDuration(s.Default); fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(defaultFieldPath, s.Default, fieldErr.Error()))
+			}
+		}
 		if s.Value != "" {
-			_, err = validateDuration(s.Value)
+			if _, fieldErr := validateDuration(s.Value); fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(valueFieldPath, s.Value, fieldErr.Error()))
+			}
 		}
 	case "delete-inactive-user-after":
+		minDeleteInactiveUserAfterErr := fmt.Errorf("must be at least %s", MinDeleteInactiveUserAfter)
+		if s.Default != "" {
+			dur, fieldErr := validateDuration(s.Default)
+			// Note: zero duration is allowed and is equivalent to "".
+			if fieldErr == nil && dur > 0 && dur < MinDeleteInactiveUserAfter {
+				fieldErr = minDeleteInactiveUserAfterErr
+			}
+			if fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(defaultFieldPath, s.Default, fieldErr.Error()))
+			}
+		}
 		if s.Value != "" {
-			var dur time.Duration
-			dur, err = validateDuration(s.Value)
-			if err == nil && dur < MinDeleteInactiveUserAfter {
-				err = fmt.Errorf("must be at least %s", MinDeleteInactiveUserAfter)
+			dur, fieldErr := validateDuration(s.Value)
+			// Note: zero duration is allowed and is equivalent to "".
+			if fieldErr == nil && dur > 0 && dur < MinDeleteInactiveUserAfter {
+				fieldErr = minDeleteInactiveUserAfterErr
+			}
+			if fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(valueFieldPath, s.Value, fieldErr.Error()))
 			}
 		}
 	case "user-last-login-default":
+		if s.Default != "" {
+			if _, fieldErr := time.Parse(time.RFC3339, s.Default); fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(defaultFieldPath, s.Default, fieldErr.Error()))
+			}
+		}
 		if s.Value != "" {
-			_, err = time.Parse(time.RFC3339, s.Value)
+			if _, fieldErr := time.Parse(time.RFC3339, s.Value); fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(valueFieldPath, s.Value, fieldErr.Error()))
+			}
 		}
 	case "user-retention-cron":
+		if s.Default != "" {
+			if _, fieldErr := cron.ParseStandard(s.Default); fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(defaultFieldPath, s.Default, fieldErr.Error()))
+			}
+		}
 		if s.Value != "" {
-			_, err = cron.ParseStandard(s.Value)
+			if _, fieldErr := cron.ParseStandard(s.Value); fieldErr != nil {
+				err = errors.Join(err, field.TypeInvalid(valueFieldPath, s.Value, fieldErr.Error()))
+			}
 		}
 	default:
 	}
 
-	if err != nil {
-		return field.TypeInvalid(field.NewPath("value"), s.Value, err.Error())
-	}
-
-	return nil
+	return err
 }
 
-func (a *admitter) validateAgentTLSMode(oldSetting, newSetting v3.Setting) error {
+func (a *admitter) validateAgentTLSMode(oldSetting, newSetting *v3.Setting) error {
```

Comment:
IMHO technically forcing dereferencing here and in `effectiveValue()` calls shouldn't have been introduced in the first place. It bears no value.  It's akin to validating pointer receivers that as a general practice under normal circumstances should not be done and is an anti-pattern. Following your logic most the code base of the webhook and rancher should be re-written to pass around values instead of references.